### PR TITLE
Feature/shipping

### PR DESCRIPTION
**Pull request by arijit1111**
Shipping Pull Request

### CODE REVIEW

File: Modules/Shipping/Http/Controllers/ShippingController.php [hunk 1]
```diff
@@ -21,48 +21,111 @@ public function __construct(ShippingService $shippingService)
...
+            return response()->json([
+                'res' => false,
+                'msg' => 'User is not authorized !',
+                'data' => ""
+            ]);
+        }
+        $response = $this->shippingService->delete($request->id);
+
         return response()->json($response);
     }
```
Comment (SonaKhachatryan1995) [inline Modules/Shipping/Http/Controllers/ShippingController.php]: Remove empty lines

File: Modules/Shipping/Http/Controllers/ShippingController.php [hunk 2]
```diff
@@ -21,48 +21,111 @@ public function __construct(ShippingService $shippingService)
...
+
+    /**
+     * Delete shipping by ID
+     *
+     * @param Request $request
+     * @return JsonResponse
+     */
+
```
Comment (SonaKhachatryan1995) [inline Modules/Shipping/Http/Controllers/ShippingController.php]: remove additional space

File: Modules/Shipping/Http/Controllers/ShippingController.php [hunk 3]
```diff
@@ -21,48 +21,111 @@ public function __construct(ShippingService $shippingService)
...
         return response()->json($response);
     }
 
-    public function update(ShippingRequest $request): JsonResponse
+    /**
+     * Update shipping details by ID
+     * @param StoreShippingRequest $request
+     * @return JsonResponse
+     */
+
```
Comment (SonaKhachatryan1995) [inline Modules/Shipping/Http/Controllers/ShippingController.php]: remove additional space

File: Modules/Shipping/Http/Controllers/ShippingController.php [hunk 4]
```diff
@@ -21,48 +21,111 @@ public function __construct(ShippingService $shippingService)
...
+        }
+
         $response = $this->shippingService->create($request->validated());
 
         return response()->json($response);
     }
 
-    public function update(ShippingRequest $request): JsonResponse
+    /**
+     * Update shipping details by ID
```
Comment (SonaKhachatryan1995) [inline Modules/Shipping/Http/Controllers/ShippingController.php:88]: add empty line below

File: Modules/Shipping/Http/Controllers/ShippingController.php [hunk 5]
```diff
@@ -21,48 +21,111 @@ public function __construct(ShippingService $shippingService)
...
+            ]);
+        }
         $response = $this->shippingService->details($request);
+
         return response()->json($response);
     }
 
     /**
-     * Store a newly created shipping in storage
+     * User can add shipping
```
Comment (SonaKhachatryan1995) [inline Modules/Shipping/Http/Controllers/ShippingController.php]: Description is not clear

File: Modules/Shipping/Http/Services/ShippingService.php [hunk 6]
```diff
@@ -52,19 +47,20 @@ public function createShipping(array $shippingData): Shipping
     }
 
     /**
-     * @param $requestData
+     * Get all shipping address by User
+     *
      * @return array
      */
```
Comment (SonaKhachatryan1995) [inline Modules/Shipping/Http/Services/ShippingService.php]: remove empty line

File: Modules/Shipping/Http/Services/ShippingService.php [hunk 7]
```diff
@@ -52,19 +47,20 @@ public function createShipping(array $shippingData): Shipping
...
 
-    public function getShippings($requestData): array
+    public function getShipping(): array
     {
```
Comment (SonaKhachatryan1995) [inline Modules/Shipping/Http/Services/ShippingService.php]: remove empty line

File: Modules/Shipping/Http/Services/ShippingService.php [hunk 8]
```diff
@@ -43,7 +38,7 @@ public function create(array $requestData): array
      */
     public function createShipping(array $shippingData): Shipping
     {
-
+        $shippingData['user_id'] = auth()->user()->id;
         $shipping = new Shipping();
         $shipping->fill($shippingData);
```
Comment (SonaKhachatryan1995) [inline Modules/Shipping/Http/Services/ShippingService.php:43]: ```$shipping->fill($shippingData);``` will fall if front end will send same id 2 times, so delete id before filling data

File: Modules/Shipping/Http/Services/ShippingService.php [hunk 9]
```diff
@@ -83,44 +79,45 @@ public function getShippings($requestData): array
...
 
         return ['res' => true, 'msg' => "", 'data' => $data];
     }
 
     /**
+     * Update shipping address BY ID
+     *
      * @param array $shippingData
      * @return array
      */
```
Comment (SonaKhachatryan1995) [inline Modules/Shipping/Http/Services/ShippingService.php]: remove empty line

**Review by SonaKhachatryan1995 — DISMISSED** [review]

File: Modules/Shipping/Http/Controllers/ShippingController.php (hunk 1 above)
Comment (arijit1111) [reply Modules/Shipping/Http/Controllers/ShippingController.php]: Done

File: Modules/Shipping/Http/Controllers/ShippingController.php (hunk 2 above)
Comment (arijit1111) [reply Modules/Shipping/Http/Controllers/ShippingController.php]: Done

File: Modules/Shipping/Http/Controllers/ShippingController.php (hunk 3 above)
Comment (arijit1111) [reply Modules/Shipping/Http/Controllers/ShippingController.php]: Done

File: Modules/Shipping/Http/Controllers/ShippingController.php (hunk 4 above)
Comment (arijit1111) [reply Modules/Shipping/Http/Controllers/ShippingController.php:88]: Done

File: Modules/Shipping/Http/Controllers/ShippingController.php (hunk 5 above)
Comment (arijit1111) [reply Modules/Shipping/Http/Controllers/ShippingController.php]: Done

File: Modules/Shipping/Http/Services/ShippingService.php (hunk 6 above)
Comment (arijit1111) [reply Modules/Shipping/Http/Services/ShippingService.php]: Done

File: Modules/Shipping/Http/Services/ShippingService.php (hunk 7 above)
Comment (arijit1111) [reply Modules/Shipping/Http/Services/ShippingService.php]: Done

File: Modules/Shipping/Http/Services/ShippingService.php (hunk 9 above)
Comment (arijit1111) [reply Modules/Shipping/Http/Services/ShippingService.php]: Done

File: Modules/Shipping/Http/Services/ShippingService.php (hunk 8 above)
Comment (arijit1111) [reply Modules/Shipping/Http/Services/ShippingService.php:43]: From fronted there is no user_id parameter pass to api

Comment (arijit1111) [reply Modules/Shipping/Http/Services/ShippingService.php:43]: Done

**Pull request merged**